notes/03-data-processing/02b-quiz-for-loops.MD

```diff
@@ -0,0 +1,132 @@
+## Quiz for-loop
+
+### With which types can we use a for-loop?
+
+1. List
+1. Map
+1. Int
+1. String
+1. Tuple2, for example (5, "Hello")
+
+#### Explanation
+
+A for-loop (without `yield`) is an alternative syntax for the method `foreach`.
+This method is available on all collections (`List`, `Array`, `Map`) and `String` 
+which is viewed as a collection of `Char` in Scala.
+
+Int and Tuple2 do not have a `foreach` method. 
+
+### What is the 4th printed line?
+
+```scala
+for {
+  width  <- List(1, 2, 3)
+  height <- List(4, 5)
+} println("area is ${width * height}")
+```
+
+1. "area is 12"
+1. Nothing, this code prints only 3 lines 
+1. "area is 5"
+1. "area is ${width * height}"
+
+#### Explanation
+
+The `s` String interpolator is missing which means `${width * height}` is not interpreted.
+
+If we add the String interpolator, then "area is 5" would be printed because the variables are evaluated as follows: 
+1. width = 1, height = 4 
+1. width = 2, height = 4 
+1. width = 3, height = 4
+1. width = 1, height = 5 
+1. width = 2, height = 5 
+1. width = 3, height = 5
```

Comment:
I think the order would be:
```suggestion
1. width = 1, height = 4 
2. width = 1, height = 5
3. width = 2, height = 4
4. width = 2, height = 5 
5. width = 3, height = 4 
6. width = 3, height = 5
```
in this case 5 is the 2nd printed line.

Reply:
You're right, thx